src/main/java/org/spongepowered/api/CatalogKey.java

```diff
@@ -119,14 +120,39 @@ static CatalogKey of(final PluginContainer container, final String value) {
     /**
      * Resolves a catalog key from a string.
      *
-     * <p>If no namespace is found in {@code string} then
+     * <p>If no namespace is found in {@code value} then
      * {@link #MINECRAFT_NAMESPACE} will be the namespace.</p>
      *
      * @param value The value
      * @return A new catalog key
      */
     static CatalogKey resolve(final String value) {
-        return Sponge.getRegistry().getBuilderRegistry().provideBuilder(Builder.class).value(value).build();
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).resolve(value, MINECRAFT_NAMESPACE);
+    }
+
+    /**
+     * Resolves a catalog key from a string, using the provided default
+     * namespace if no namespace was found in {@code value}.
+     *
+     * @param value The value
+     * @param defaultNamespace The default namespace
+     * @return A new catalog key
+     */
+    static CatalogKey resolve(final String value, final String defaultNamespace) {
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).resolve(value, defaultNamespace);
+    }
+
+    /**
+     * Resolves a catalog key from a string, using the provided
+     * {@link PluginContainer}'s identifier as the default namespace if no
+     * namespace was found in {@code value}.
+     *
+     * @param value The value
+     * @param defaultNamespace The default namespace
+     * @return A new catalog key
+     */
+    static CatalogKey resolve(final String value, final PluginContainer defaultNamespace) {
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).resolve(value, defaultNamespace);
```

Comment:
```suggestion
        return resolve(value, defaultNamespace.getMetadata().getId());
```
IMO we shouldn't hide the logic in the factory as it is clearly defined in the javadocs.